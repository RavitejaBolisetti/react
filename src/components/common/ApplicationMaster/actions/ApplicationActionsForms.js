/*
 *   Copyright (c) 2023 Mahindra & Mahindra Ltd. 
 *   All rights reserved.
 *   Redistribution and use of any source or binary or in any form, without written approval and permission is prohibited. Please read the Terms of Use, Disclaimer & Privacy Policy on https://www.mahindra.com/
 */
import React from 'react';
import { Input, Form, Col, Row, Switch, Button, Select } from 'antd';
import { PlusOutlined } from '@ant-design/icons';

import { duplicateValidator, validateRequiredSelectField } from 'utils/validation';
import { preparePlaceholderSelect } from 'utils/preparePlaceholder';

const ApplicationActionsForm = ({ finalFormdata, form, onFinish, status, name, id, isEditing, isBtnDisabled, actions,disableStatus, setIsBtnDisabled, onFieldsChange }) => {
    const onFinishFailed = (err) => {
        console.error(err);
    };

    const fieldNames = { label: 'actionName', value: 'id' };

    return (
        <>
            <Row gutter={20}>
                <Col xs={24} sm={24} md={24} lg={24} xl={24} xxl={24}>
                    <Form form={form} onFieldsChange={onFieldsChange} autoComplete="off" id="applicationActionsForm" layout="vertical" onFinish={onFinish} onFinishFailed={onFinishFailed}>
                        <Row gutter={20}>
                            <Col xs={12} sm={12} md={12} lg={12} xl={12} xxl={12}>
                                <Form.Item label="Action" name="applicationName" rules={[validateRequiredSelectField('application action'),  { validator: (rule, value) => duplicateValidator(value?.label, 'actionName', finalFormdata?.applicationAction)   }]}>  
                                    <Select
                                        getPopupContainer={(triggerNode) => triggerNode.parentElement}
                                        labelInValue
                                        placeholder={preparePlaceholderSelect('application action')}
                                        fieldNames={fieldNames}
                                        options={actions?.filter((el) => el?.actionName)}
                                        disabled={isBtnDisabled || disableStatus}
                                        allowClear
                                    />
                                </Form.Item>
                            </Col>
                            <Col xs={12} sm={12} md={12} lg={12} xl={12} xxl={12}>
                                <Form.Item onClick={e => e.stopPropagation()} initialValue={true} labelAlign="left" wrapperCol={{ span: 24 }} name="status" label="Status" valuePropName="checked">
                                    <Switch disabled={isBtnDisabled} checkedChildren="Active" unCheckedChildren="Inactive" valuePropName="checked" />
                                </Form.Item>
                            </Col>
                            {!isEditing && (
                                <Col xs={4} sm={4} md={4} lg={4} xl={4} xxl={4}>
                                    <Button form="applicationActionsForm" disabled={isBtnDisabled} icon={<PlusOutlined />} htmlType="submit" type="primary">
                                        Add
                                    </Button>
                                </Col>
                            )}
                            <Form.Item name='id' hidden>
                                <Input />
                            </Form.Item>
                            <Form.Item name='actionMasterId' hidden>
                                <Input />
                            </Form.Item>
                        </Row>
                    </Form>
                </Col>
            </Row>
        </>
    );
};

export default ApplicationActionsForm;
