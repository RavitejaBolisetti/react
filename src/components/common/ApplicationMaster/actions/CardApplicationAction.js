import React, { useState } from 'react';
import { Col, Card, Row, Button, Divider, Form } from 'antd';
import { FiEdit, FiTrash } from 'react-icons/fi';
import { Typography } from 'antd';

import style from './../../Common.module.css';


import ApplicationActionsForm from './ApplicationActionsForms';
import { Fragment } from 'react';

const { Text } = Typography;

const CardApplicationAction = (props) => {
    const { status, actionId, actionName, actionMasterId, id, setFinalFormdata, forceUpdate, setIsBtnDisabled, isBtnDisabled, actions } = props;
    const [form] = Form.useForm();
    const [isEditing, setIsEditing] = useState(false);

    const onEdit = (values) => {
        form.setFieldsValue({
            applicationName: {
                label: values.actionName,
                value: values.actionMasterId,
                id: values.actionMasterId,
            },
            actionMasterId: actionMasterId,
            status: values.status,
            id: id,
        });
        setIsEditing(true);
        setIsBtnDisabled(true);
    };

    const onUpdate = () => {

        const newFormData = form.getFieldsValue();
        setFinalFormdata((prev) => {
            const newList = prev?.applicationAction;
            const index = prev?.applicationAction?.findIndex((el) => el.actionId === actionId);
            const prevData = prev?.applicationAction[index];
            newList.splice(index, 1, { ...prevData, status: newFormData?.status });
            return { ...prev, applicationAction: [...newList] };
        });
        setIsEditing(false);
        setIsBtnDisabled(false);
        form.resetFields();
        forceUpdate();
    };

    const handleDeleteAction = (val) => {
        setFinalFormdata((prev) => {
            const newList = prev?.applicationAction;
            const indx = prev?.applicationAction?.findIndex((el) => el?.actionId === val.actionId);
            newList.splice(indx, 1);
            return { ...prev, applicationAction: newList };
        });

        setIsEditing(false);
        setIsBtnDisabled(false);
        form.resetFields();
    };

    const onCancel = () => {
        setIsEditing(false);
        setIsBtnDisabled(false);
    };

    return (
        <>
            <Card
               className={style.viewCardSize}  
               key={actionId}                   
            >
                <Row align="middle">
                    <Col xs={18} sm={18} md={18} lg={18} xl={18} xxl={18}>
                        <Col xs={16} sm={16} md={16} lg={16} xl={16} xxl={16}>
                            <Text type="secondary">Status: </Text> {status ? <Text type="success">Active</Text> : <Text type="secondary">Inactive</Text>}
                        </Col>
                        <Col xs={16} sm={16} md={16} lg={16} xl={16} xxl={16}>
                            <Text strong>{actionName}</Text>
                        </Col>
                        <Col xs={16} sm={16} md={16} lg={16} xl={16} xxl={16}>
                            <Text type="secondary">Action ID: {actionId}</Text>
                        </Col>
                    </Col>
                    <Col xs={6} sm={6} md={6} lg={6} xl={6} xxl={6}>
                        <Row justify="end">
                            {!isEditing ? (
                                <>
                                    <Col xs={6} sm={6} md={6} lg={6} xl={6} xxl={6}>
                                        <Button disabled={isBtnDisabled} type="link" icon={<FiEdit />} onClick={() => onEdit({ status, actionName, actionId, actionMasterId, id })} />
                                    </Col>
                                    {!id ?
                                        <Col xs={6} sm={6} md={6} lg={6} xl={6} xxl={6}>
                                            <Button disabled={isBtnDisabled} onClick={() => handleDeleteAction({ status, actionName, actionId })} type="link" icon={<FiTrash />}></Button>
                                        </Col>
                                        : ''}
                                </>
                            ) : (
                                <>
                                    {' '}
                                    <Col xs={12} sm={12} md={12} lg={12} xl={12} xxl={12}>
                                        <Button type="link" onClick={onUpdate}>
                                            Save
                                        </Button>
                                    </Col>
                                    <Col xs={12} sm={12} md={12} lg={12} xl={12} xxl={12}>
                                        <Button type="link" onClick={() => onCancel()}>
                                            Cancel
                                        </Button>
                                    </Col>
                                </>
                            )}
                        </Row>
                    </Col>
                </Row>

                {isEditing && (
                    <Fragment>
                        <Divider />
                        <ApplicationActionsForm status={status} name={actionName} id={id} form={form} isEditing={isEditing} actions={actions} disableStatus={true} />
                    </Fragment>
                )}
            </Card>
        </>
    );
};

export default CardApplicationAction;
