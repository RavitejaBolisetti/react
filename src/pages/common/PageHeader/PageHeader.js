import React from 'react';
import { Row, Col, Space, Button, Modal } from 'antd';
import { useNavigate } from 'react-router-dom';
import { ExclamationCircleFilled } from '@ant-design/icons';
import { FaHeart, FaHistory, FaLongArrowAltLeft, FaRegHeart } from 'react-icons/fa';

import { addToolTip } from 'utils/customMenuLink';
import { ROUTING_DASHBOARD } from 'constants/routing';
import styles from './PageHeader.module.css';

const { confirm } = Modal;

export const PageHeader = ({ pageTitle, isFavourite, handleFavouriteClick, visibleChangeHistory = true, handleChangeHistoryClick = undefined, isChangeHistoryVisible = false }) => {
    const navigate = useNavigate();
    const handleBack = () => {
        confirm({
            title: 'Are you sure to leave this page?',
            icon: <ExclamationCircleFilled />,
            content: 'If you leave this page, All unsaved data will be lost',
            okText: 'Yes',
            okType: 'danger',
            cancelText: 'No',
            cancelType: 'danger',
            onOk() {
                navigate(-1) || navigate(ROUTING_DASHBOARD);
            },
            onCancel() {},
        });
    };

    return (
        <>
            <Row gutter={20}>
                <Col xs={16} sm={24} md={12} lg={18} xl={18} xxl={18} className={styles.padRight0}>
                    <Space>
                        <div>
                            <span className={styles.headingGradient}>{pageTitle}</span>
                        </div>
                        <div className={styles.favIconHeading}>{isFavourite ? addToolTip('Mark as unfavourite')(<FaHeart color="#ff3e5b" size={18} onClick={handleFavouriteClick} />) : addToolTip('Mark as favourite')(<FaRegHeart size={18} onClick={handleFavouriteClick} />)}</div>
                    </Space>
                </Col>
                <Col xs={8} sm={24} md={12} lg={6} xl={6} xxl={6} className={styles.padRight0}>
                    <div className={styles.buttonContainer}>
                        {visibleChangeHistory &&
                            (isChangeHistoryVisible ? (
                                <Button type="primary" onClick={handleChangeHistoryClick}>
                                    <FaHistory className={styles.buttonIcon} />
                                    Change History
                                </Button>
                            ) : (
                                <Button danger onClick={handleChangeHistoryClick}>
                                    <FaHistory className={styles.buttonIcon} />
                                    Change History
                                </Button>
                            ))}

                        <Button danger onClick={handleBack} className="btn btn-outline srchbtn mr0 boxShdwNon">
                            <FaLongArrowAltLeft className={styles.buttonIcon} />
                            Exit
                        </Button>


                       

                    </div>
                </Col>
            </Row>

            <Row gutter={20}>
                <Col xs={24} sm={24} md={12} lg={24} xl={24} xxl={24}>
                    <div className={styles.pageHeaderNameSection}></div>
                </Col>
            </Row>
        </>
    );
};
