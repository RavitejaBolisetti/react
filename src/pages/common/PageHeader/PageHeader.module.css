.buttonContainer :global .ant-btn-default.ant-btn-dangerous:not(:disabled):hover {
    background-color: #ff3e5b;
    color: #ffffff !important;
}

:global .ant-btn {
    height: 34px;
    padding: 6px 12px;
    border-radius: 4px;
}

.padRight0 {
    padding-right: 0px !important;
}

.pageHeader .headingGradient {
    padding-bottom: 10px;
    border-radius: 5px;
    border: 0px;
    border-bottom: 4px solid;
    border-image: linear-gradient(to right, #efa3a3, #cac9ee);
    border-image-slice: 1;
    text-decoration: none;
    margin-bottom: 28px;
    color: #0b0b0c;
    font-size: 20px;
}

.pageHeader .favIconHeading {
    padding-left: 5px;
    margin-top: 5px;
    line-height: 4px;
    padding-top: 3px;
    cursor: pointer;
    color: #ff3e5b;
}

.pageHeaderNameSection {
    border-bottom: 1px solid #dedede;
    margin: 0 0 16px 0;
}

.pageHeader .buttonContainer {
    text-align: right;
}

.pageHeader .buttonContainer button {
    margin-left: 14px;
}

.pageHeader .buttonContainer .buttonIcon {
    margin-right: 6px;
    padding-top: 2px;
}

/* Modal Content */

:global .ant-modal-confirm .ant-modal-confirm-body .ant-modal-confirm-title {
    display: contents;
}

:global .ant-modal-confirm .ant-modal-confirm-btns {
    text-align: center;
    margin-top: 0;
}

:global .ant-modal-confirm .ant-modal-confirm-body .ant-modal-confirm-title+.ant-modal-confirm-content {
    max-width: 100%;
    padding: 10px;
    text-align: justify;
    border-top: 1px solid #DEDEDE;
}

:global .ant-modal-confirm .ant-modal-confirm-btns button:last-child {
    background-color: #ff3e5b;
    color: #ffffff !important;
}

:global .ant-modal-confirm .ant-modal-confirm-btns button {
    text-align: center;
    width: 100px;
    margin-top: 15px;
    background-color: #ffffff;
    border: 1px solid #ff3e5b;
    color: #ff3e5b !important;
}

:global .ant-modal-confirm .ant-modal-confirm-btns button:hover {
    background-color: #ff3e5b;
    border: 1px solid #ff3e5b;
    color: #ffffff !important;
}

.modalIconAlert {
    color: #FF7A00;
    padding-right: 8px;
}

@media (max-width: 768px) {

    .pageHeader .headingGradient {
        padding-bottom: 5px;
        /* border-bottom: 0px; */
        font-size: 14px;
        font-weight: 600;
    }

    .pageHeaderNameSection {
        border-bottom: 0;
        padding: 0px;
        margin: 0 0 10px 0;
        /* width: 68vw; */
    }

    .pageHeader .buttonContainer {
        margin-top: 6px;
    }

    .pageHeader .buttonContainer button {
        margin-left: 5px;
    }

    .pageHeader .buttonContainer :global .ant-btn {
        font-size: 12px;
        height: 24px;
        padding: 2px 6px;
    }

    .pageHeader .buttonContainer .buttonIcon {
        margin-right: 2px;
    }

    .buttonContainer button {
        margin-left: 0px;
    }

}