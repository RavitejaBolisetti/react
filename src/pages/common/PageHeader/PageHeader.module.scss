@import 'assets/sass/app.module.scss';

.pageHeader {
    height: 37px;

    & :global {
        .ant-btn {
            height: 34px;
            @include space(padding, top, $sp4);
            @include space(padding, right, $sp12);
            @include space(padding, bottom, $sp4);
            @include space(padding, left, $sp12);
            @include borderRedius(4px, 1px, solid, inherit);
        }
    }

    .headingGradient {
        //padding-bottom: 12px;
        @include space(padding, bottom, $sp13);
        // @include space(margin, bottom, $sp28);
        @include borderRedius(4px, 0px, solid, inherit);
        border-bottom: 4px solid;
        @include bordergradient($DarkPink, $LightPink);
        border-image-slice: 1;
        @include font-style(16, 24, 700, $primaryBlack, $fontfamily, left);
        text-decoration: none;

        @include breakpoint(md) {
            padding-bottom: 5px;
            @include space(padding, bottom, $sp5);
            @include font-style(14, 24, 600, $primaryBlack, $fontfamily, left);
        }

        .favIconHeading {
            @include space(padding, top, $sp3);
            @include space(margin, top, $sp5);
            @include space(margin, left, $sp14);
            line-height: 4px;
            cursor: pointer;
            color: $primaryRed;
        }
    }

    .buttonContainer {
        text-align: right;

        @include breakpoint(md) {
            @include space(margin, top, $sp6);
        }

        & :global {
            .ant-btn {
                @include breakpoint(md) {
                    //font-size: 12px;
                    @include font-style(12, inherit, normal, normal, $fontfamily, right);
                    height: 24px;
                    @include space(padding, top, $sp2);
                    @include space(padding, right, $sp6);
                    @include space(padding, bottom, $sp2);
                    @include space(padding, left, $sp6);
                }
            }

            .ant-btn-default {
                &.ant-btn-dangerous {
                    &:not(:disabled):hover {
                        background-color: $primaryRed;
                        color: $primaryWhite !important;
                    }
                }
            }
        }

        button {
            margin-left: 14px;

            @include breakpoint(md) {
                @include space(margin, left, $sp5);
                // margin-left: 0px;
            }
        }

        .buttonIcon {
            @include space(padding, top, $sp2);
            @include space(margin, right, $sp6);

            @include breakpoint(md) {
                @include space(margin, right, $sp2);
            }
        }
    }
}

.pageHeaderNameSection {
    border-bottom: 1px solid $neutralGrey1;
    @include space(margin, top, $sp1);
    @include space(margin, bottom, $sp20);

    @include breakpoint(md) {
        border-bottom: 0;
        @include space(padding, all, $sp0);
        @include space(margin, top, $sp0);
        @include space(margin, right, $sp0);
        @include space(margin, bottom, $sp10);
        @include space(margin, left, $sp0);
    }
}

/* Modal Content */
:global {
    .ant-modal-confirm {
        .ant-modal-confirm-body {
            .ant-modal-confirm-title {
                display: contents;
            }

            .ant-modal-confirm-title + .ant-modal-confirm-content {
                max-width: 100%;
                padding: 10px;
                text-align: justify;
                border-top: 1px solid $neutralGrey1;
            }
        }

        .ant-modal-confirm-btns {
            text-align: center;
            margin-top: 0;

            button {
                &:last-child {
                    background-color: $primaryRed;
                    color: $primaryWhite !important;
                }

                text-align: center;
                width: 100px;
                margin-top: 15px;
                background-color: $primaryWhite;
                border: 1px solid $primaryRed;
                color: $primaryRed !important;

                &:hover {
                    background-color: $primaryRed;
                    border: 1px solid $primaryRed !important;
                    color: $primaryWhite !important;
                }
            }
        }
    }
}
