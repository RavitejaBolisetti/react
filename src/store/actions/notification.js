/*
 *   Copyright (c) 2023 Mahindra & Mahindra Ltd. 
 *   All rights reserved.
 *   Redistribution and use of any source or binary or in any form, without written approval and permission is prohibited. Please read the Terms of Use, Disclaimer & Privacy Policy on https://www.mahindra.com/
 */
export const NOTIFICATION_SHOW = 'NOTIFICATION_SHOW';
export const NOTIFICATION_HIDE = 'NOTIFICATION_HIDE';

export const showGlobalNotification = ({ notificationType = 'error', title, message, placement, showTitle }) => ({
    type: NOTIFICATION_SHOW,
    visible: true,
    notificationType,
    title,
    message,
    placement,
    showTitle,
});

export const hideGlobalNotification = (data) => ({
    type: NOTIFICATION_HIDE,
    visible: false,
});
