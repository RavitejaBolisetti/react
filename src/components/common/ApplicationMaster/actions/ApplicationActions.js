import React, { Fragment, useState, useReducer } from 'react';
import { Form, Space, Divider } from 'antd';

import CardApplicationAction from './CardApplicationAction';
import ApplicationActionsForm from './ApplicationActionsForms';

const ApplicationActions = ({ footerEdit = false, onFinishFailed = () => {}, isReadOnly = false, setFormBtnDisable, setFinalFormdata, finalFormdata, actions }) => {
    const [, forceUpdate] = useReducer((x) => x + 1, 0);
    const [isBtnDisabled, setIsBtnDisabled] = useState(false);
    const [actionForm] = Form.useForm();

    const onActionFormFinish = (val) => {
        const { value, label } = val?.applicationName;
        const { actionId, ...selectedActionData } = actions?.find((el) => el?.id === value);
        setFinalFormdata({ ...finalFormdata, applicationAction: [...finalFormdata.applicationAction, { actionName: label, status: val?.status, id: val?.id, actionMasterId: value, actionId: actionId }] });
        actionForm.resetFields();
    };

    return (
        <Fragment>
            <Divider/>
            <div style={{ marginBottom: '14px' }}>
                <ApplicationActionsForm finalFormdata={finalFormdata} form={actionForm} onFinish={onActionFormFinish} setIsBtnDisabled={setIsBtnDisabled} isBtnDisabled={isBtnDisabled} actions={actions} />
            </div>
            {finalFormdata?.applicationAction?.length > 0 &&
                finalFormdata?.applicationAction?.map((action) => {
                    return <CardApplicationAction {...action} form={actionForm} onFinish={onActionFormFinish} setFinalFormdata={setFinalFormdata} forceUpdate={forceUpdate} setIsBtnDisabled={setIsBtnDisabled} isBtnDisabled={isBtnDisabled} actions={actions} />;
                })}
        </Fragment>
    );
};

export default ApplicationActions;
