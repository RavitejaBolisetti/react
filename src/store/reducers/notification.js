/*
 *   Copyright (c) 2023 Mahindra & Mahindra Ltd. 
 *   All rights reserved.
 *   Redistribution and use of any source or binary or in any form, without written approval and permission is prohibited. Please read the Terms of Use, Disclaimer & Privacy Policy on https://www.mahindra.com/
 */
import { NOTIFICATION_SHOW, NOTIFICATION_HIDE } from 'store/actions/notification';

const initialState = {
    visible: false,
    showTitle: true,
    notificationType: 'error',
    title: 'ERROR',
    message: null,
    placement: null,
};

export const notification = (state = initialState, action) => {
    switch (action.type) {
        case NOTIFICATION_SHOW:
            return { ...state, visible: true, notificationType: action.notificationType, title: action.title, message: action.message, placement: action.placement,showTitle: action.showTitle };
        case NOTIFICATION_HIDE:
            return { ...initialState };
        default:
            return { ...state };
    }
};
