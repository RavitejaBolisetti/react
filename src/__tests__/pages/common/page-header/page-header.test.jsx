/*
 *   Copyright (c) 2023 Mahindra & Mahindra Ltd. 
 *   All rights reserved.
 *   Redistribution and use of any source or binary or in any form, without written approval and permission is prohibited. Please read the Terms of Use, Disclaimer & Privacy Policy on https://www.mahindra.com/
 */
import '@testing-library/jest-dom/extend-expect';
import customRender from '@utils/test-utils';
import PageHeader from '@pages/common/PageHeader/PageHeader';
import { screen, fireEvent } from '@testing-library/react';
import { showGlobalNotification } from 'store/actions/notification';
const mockFetchList = jest.fn();
const mockMarkFavourite = jest.fn();
const mockListShowLoading = jest.fn();
const mockShowGlobalNotification = jest.fn();

describe('PageHeader Component', () => {
    it('should render PageHeader component', async () => {
        customRender(<PageHeader isFavourite={false} canMarkFavourite={true} mockFetchList={mockFetchList} mockMarkFavourite={mockMarkFavourite} mockShowGlobalNotification={mockShowGlobalNotification} mockListShowLoading={mockListShowLoading} />);
    });

    it('renders page title correctly', () => {
        customRender(<PageHeader pageTitle="Test Page" />);
        const pageTitleElement = screen.getByText('Test Page');
        expect(pageTitleElement).toBeInTheDocument();
    });

    it('displays "Mark as favorite" icon when canMarkFavourite is true', () => {
        customRender(<PageHeader pageTitle="Mark as favourite" canMarkFavourite={true} onError={jest.fn()} handleFavouriteClick={jest.fn()} isFavourite={false} />);
        const markAsFavoriteIcon = screen.getByTestId('addfav');
        expect(markAsFavoriteIcon).toBeInTheDocument();
    });

    it('displays "Mark as favorite" icon when canMarkFavourite is true', async () => {
        customRender(<PageHeader pageTitle="Mark as favourite" canMarkFavourite={true} onSuccess={jest.fn()} onError={jest.fn()} handleFavouriteClick={jest.fn()} isFavourite={false} showGlobalNotification={showGlobalNotification} />);
        const markAsFavoriteIcon = screen.getByTestId('addfav');
        fireEvent.click(markAsFavoriteIcon);
    });

    it('calls handleFavouriteClick when "Mark as favorite" icon is clicked', () => {
        customRender(<PageHeader pageTitle="Test Page" canMarkFavourite={true} onError={jest.fn()} handleFavouriteClick={jest.fn()} isFavourite={true} />);
        const markAsFavoriteIcon = screen.getByTestId('addfav');
        fireEvent.click(markAsFavoriteIcon);
    });
});
