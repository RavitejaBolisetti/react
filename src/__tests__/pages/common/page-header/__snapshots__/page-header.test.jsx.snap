// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Component should render PageHeader component 1`] = `
<div>
  <div
    class="ant-row pageHeader css-dev-only-do-not-override-14wwjjs"
    style="margin-left: -10px; margin-right: -10px;"
  >
    <div
      class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 ant-col-xxl-24 css-dev-only-do-not-override-14wwjjs"
      style="padding-left: 10px; padding-right: 10px;"
    >
      <div
        class="ant-space css-dev-only-do-not-override-14wwjjs ant-space-horizontal ant-space-align-center"
      >
        <div
          class="ant-space-item"
        >
          <div>
            <span
              class="headingGradient"
            >
              <span
                class="favIconHeading"
              >
                <svg
                  baseProfile="tiny"
                  color="#B5B5B6"
                  data-testid="addfav"
                  fill="currentColor"
                  height="13"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(181, 181, 182);"
                  version="1.2"
                  viewBox="0 0 24 24"
                  width="13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.1 11.3l3.6 3.3-1 4.6c-.1.6.1 1.2.6 1.5.2.2.5.3.8.3.2 0 .4 0 .6-.1 0 0 .1 0 .1-.1l4.1-2.3 4.1 2.3s.1 0 .1.1c.5.2 1.1.2 1.5-.1.5-.3.7-.9.6-1.5l-1-4.6c.4-.3 1-.9 1.6-1.5l1.9-1.7.1-.1c.4-.4.5-1 .3-1.5s-.6-.9-1.2-1h-.1l-4.7-.5-1.9-4.3s0-.1-.1-.1c-.1-.7-.6-1-1.1-1-.5 0-1 .3-1.3.8 0 0 0 .1-.1.1l-1.9 4.3-4.7.5h-.1c-.5.1-1 .5-1.2 1-.1.6 0 1.2.4 1.6z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row css-dev-only-do-not-override-14wwjjs"
    style="margin-left: -10px; margin-right: -10px;"
  >
    <div
      class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 ant-col-xxl-24 css-dev-only-do-not-override-14wwjjs"
      style="padding-left: 10px; padding-right: 10px;"
    >
      <div
        class="pageHeaderNameSection"
      />
    </div>
  </div>
  ;
</div>
`;
